### ISSUE TITLE
[a11y] Crash in Android accessibility bridge when using range slider

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#60926]
b/161765861

Run the following code on Android with TalkBack enabled (tested at commit 4b12050112afd581ddf53df848275fa681f908f3):

```dart
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Canvas Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Canvas Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  final String title;

  MyHomePage({Key key, this.title}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  RangeValues _currentRangeValues = const RangeValues(40, 80);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: RangeSlider(
          values: _currentRangeValues,
          min: 0,
          max: 100,
          divisions: 5,
          labels: RangeLabels(
            _currentRangeValues.start.round().toString(),
            _currentRangeValues.end.round().toString(),
          ),
          onChanged: (RangeValues values) {
            setState(() {
              _currentRangeValues = values;
            });
          },
        ),
      ),
    );
  }
}
```

See the following exception:

```
E/flutter ( 9096): [ERROR:flutter/shell/platform/android/platform_view_android_jni_impl.cc(43)] java.lang.ArrayIndexOutOfBoundsException: length=2; index=2
E/flutter ( 9096): 	at io.flutter.view.AccessibilityBridge.updateCustomAccessibilityActions(AccessibilityBridge.java:1262)
E/flutter ( 9096): 	at io.flutter.view.AccessibilityBridge$1.updateCustomAccessibilityActions(AccessibilityBridge.java:260)
E/flutter ( 9096): 	at io.flutter.embedding.engine.FlutterJNI.updateCustomAccessibilityActions(FlutterJNI.java:499)
E/flutter ( 9096): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/flutter ( 9096): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/flutter ( 9096): 	at android.os.Looper.loop(Looper.java:174)
E/flutter ( 9096): 	at android.app.ActivityThread.main(ActivityThread.java:7356)
E/flutter ( 9096): 	at java.lang.reflect.Method.invoke(Native Method)
E/flutter ( 9096): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
E/flutter ( 9096): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
E/flutter ( 9096): 
F/flutter ( 9096): [FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(977)] Check failed: CheckException(env). 
F/libc    ( 9096): Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 9096 (e.animated_size), pid 9096 (e.animated_size)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'google/flame/flame:10/QQ2A.200405.005/6254899:user/release-keys'
Revision: 'MP1.0'
ABI: 'arm64'
Timestamp: 2020-07-06 13:07:17-0700
pid: 9096, tid: 9096, name: e.animated_size  >>> com.example.animated_size <<<
uid: 10216
signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: '[FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(977)] Check failed: CheckException(env). 
'
    x0  0000000000000000  x1  0000000000002388  x2  0000000000000006  x3  0000007fd2f48ff0
    x4  0080000000000000  x5  0080000000000000  x6  0080000000000000  x7  0000000000008000
    x8  00000000000000f0  x9  38ea7c7919251262  x10 0000000000000001  x11 0000000000000000
    x12 fffffff0fffffbdf  x13 0000000004fb2710  x14 0000000000000001  x15 000000000002ad98
    x16 0000007d5d0808c0  x17 0000007d5d05e0c0  x18 0000007d5ecd8000  x19 0000000000002388
    x20 0000000000002388  x21 00000000ffffffff  x22 0000007d5e2f8398  x23 00000000ffffffff
    x24 0000000000000018  x25 0000000000000018  x26 000000000000012f  x27 0000000000000115
    x28 000000000000002f  x29 0000007fd2f49090
    sp  0000007fd2f48fd0  lr  0000007d5d012f48  pc  0000007d5d012f74
backtrace:
      #00 pc 0000000000081f74  /apex/com.android.runtime/lib64/bionic/libc.so (abort+160) (BuildId: 8de865099c99977483c8947f9b7937e9)
      #01 pc 0000000001257fcc  /data/app/com.example.animated_size-byfNU9hTH-fjGfIDqgOoXQ==/lib/arm64/libflutter.so (BuildId: df9d67f61ebd66906967bce5885cff2f6c89d58e)
      #02 pc 00000000000019dc  [anon:.bss]

```

**Comment by goderbauer** [comment]
The code used is taken from the code sample of RangeSlider on our API docs: https://master-api.flutter.dev/flutter/material/RangeSlider-class.html

**Comment by chunhtai** [comment]
https://github.com/flutter/engine/blob/9b3e3410f02681e446ca48038164a553ababb143/shell/platform/android/platform_view_android.cc#L393
This should use `action_strings` instead of `string`. It is introduced in https://github.com/flutter/engine/pull/18903/files

**Comment by njovy** [comment]
@chunhtai This error occurs when I use Firebase ML vision plugin on Android. Will this fix gets cherry picked to 1.20?

**Comment by njovy** [comment]
@chunhtai This bug appears in 1.20.0 as well yet this patch wasn't included to 1.20.0-7.4.pre.

**Comment by chunhtai** [comment]
@pcsosinski This is a rather nasty regression, it will crash android for a wide range of widget that uses custom actions. Can we consider cherry picking this into the next release?

**Comment by pcsosinski** [comment]
@chunhtai have you verified it on master?

**Comment by chunhtai** [comment]
yes, it works in master, this engine roll contains the fix https://github.com/flutter/flutter/commit/2273961720838f52841952b471c20d1b6b2a3b80
